### ISSUE TITLE
twisted.internet.test.test_fdset fails on OpenBSD because connect() succeeds

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4206]
|<img alt="soyt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| soyt reported|
|-|-|
|Trac ID|trac#4206|
|Type|defect|
|Created|2010-01-12 10:24:25Z|
|Branch|https://github.com/twisted/twisted/tree/synchronous-nonblocking-connect-4206|

On OpenBSD (at least), calling connect() on non-blocking sockets may succeed immediatly (without setting errno to EINPROGESS) if the remote end is local. This makes a bunch of test fail.
Suggested patch fixes it.

Attachments:

* [patch-twisted_internet_test_test_fdset_py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d6b/d6b921ce93f6474cbbf71f0d9649500cd39e7e41/dc9645b46afaf71c92803e5aa720f283db462b30) (689 bytes) - added by soyt on 2010-01-12 10:24:57Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4206 4206
type__defect defect
reporter__soyt soyt
priority__normal normal
milestone__ 
branch__branches_synchronous_nonblocking_connect_4206 branches/synchronous-nonblocking-connect-4206
branch_author__soyt soyt
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1263291865000000 1263291865000000
changetime__1272845812000000 1272845812000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

Thanks for the patch.  Just adjusting metadata (OpenBSD isn't a supported platform, so we can't have regressions there, only defects ;).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

For what it's worth, I'm pretty sure I remember that this is a problem on Solaris as well.  Unfortunately the Solaris slave is offline so I can't verify that, or check to see if this patch corrects the issue.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [28882]) Branching to 'synchronous-nonblocking-connect-4206'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

FreeBSD 8 seems to have this behavior as well.  And we now have some FreeBSD 8 slaves.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [28883]) Apply patch from soyt to allow connect to succeed

refs [#4206](https://github.com/twisted/twisted/issues/4206)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

The patch fixes a couple dozen tests on all of the FreeBSD 8 builders.  That seems to be a good reason to apply it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [28888]) Merge synchronous-nonblocking-connect-4206

Author: soyt
Reviewer: exarkun
Fixes: [#4206](https://github.com/twisted/twisted/issues/4206)

Change the `IReactorFDSet` tests to allow for non-blocking connections
to succeed synchronously.  This happens on various BSDs for connections
to localhost.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
